### PR TITLE
New hash functions BLAKE2s, BLAKE2b, Groestl256 and Groestl512

### PR DESCRIPTION
**Pull request by rfswarm**
Implementation of the sha3 hash function finalists BLAKE and Groestl.

### CODE REVIEW

File: sys/include/crypto/blake2b.h [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    uint8_t  buf[2 * BLAKE2B_BLOCKBYTES];
+    size_t   buflen;
+    uint8_t  last_node;
+} blake2b_state;
+
+/**
+ * @name Algorithm macros
+ * @{
+ */
+#define G(r,i,a,b,c,d) \
```
Comment (Kijewski) [inline sys/include/crypto/blake2b.h]: Please put `G`and `BLAKE2B_ROUND` into the .c file.

File: sys/include/crypto/blake2s.h [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    uint8_t  buf[2 * BLAKE2S_BLOCKBYTES];
+    size_t   buflen;
+    uint8_t  last_node;
+} blake2s_state;
+
+/**
+ * @name Algorithm macros
+ * @{
+ */
+#define G(r,i,a,b,c,d) \
```
Comment (Kijewski) [inline sys/include/crypto/blake2s.h]: Please put `G`and `BLAKE2S_ROUND` into the .c file.

**Comment by rfswarm** [comment]
Fixing done and ready for more, fixing some final doxygen bugs releated to typedef(d) structs soon

File: sys/include/crypto/groestl512.h [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+/** @} */
+
+/**
+ * @name Algorithm macros
+ * @{
+ */
+#define mul2(x,t) \
+{\
+  t = x & 0x80808080;\
+  x ^= t;\
```
Comment (OlegHahm) [inline sys/include/crypto/groestl512.h]: indentation

**Comment by OlegHahm** [comment]
Unittests for Arduino Mega are not building.

Also, please prefix the commit message.

**Comment by Kijewski** [comment]
This PR appears to be an astyled extract of https://github.com/dchest/pyblake2/blob/master/impl/blake2b.c and https://github.com/dchest/pyblake2/blob/e8c87deccda3331f84426a57b0ceb6cf3184e12f/impl/blake2-impl.h. The original author put the code into the public domain, so the relicensing is OK.

**Comment by rfswarm** [comment]
?! its based on supercop ref impl - cant remember which exact version...

**Comment by OlegHahm** [comment]
Apparently abandoned.

**Pull request closed without merging**